Assets/Plugins/iOS/Shopify/ApplePayEventResponse.swift

```diff
@@ -47,43 +47,37 @@ struct ApplePayEventResponse {
     // ----------------------------------
     //  MARK: - Init -
     //
-    init?(serialized: String) {
-
-        guard
-            let data = serialized.data(using: .utf8),
-            let jsonObject = (try? JSONSerialization.jsonObject(with: data)) as? JSON else {
-                return nil
-        }
-
+    static func deserialized(_ serialized: JSON) -> ApplePayEventResponse? {
```

Comment:
Should we not be using the `-ed` ending here since it is a nonmutating method?
[Side Effect Naming](https://swift.org/documentation/api-design-guidelines/#name-according-to-side-effects)